planner/python_poetry_planner.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 Jetpack Technologies Inc and contributors. All rights reserved.
+// Use of this source code is governed by the license in the LICENSE file.
+
+package planner
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pelletier/go-toml/v2"
+)
+
+type PythonPoetryPlanner struct{}
+
+// PythonPoetryPlanner implements interface Planner (compile-time check)
+var _ Planner = (*PythonPoetryPlanner)(nil)
+
+func (g *PythonPoetryPlanner) Name() string {
+	return "PythonPoetryPlanner"
+}
+
+func (g *PythonPoetryPlanner) IsRelevant(srcDir string) bool {
+	poetryLockPath := filepath.Join(srcDir, "poetry.lock")
+	mainPYPath := filepath.Join(srcDir, "main.py")
+	return fileExists(poetryLockPath) && fileExists(mainPYPath)
+}
+
+func (g *PythonPoetryPlanner) GetPlan(srcDir string) *Plan {
+	version := g.PythonVersion(srcDir)
+	return &Plan{
+		Packages: []string{
+			fmt.Sprintf("python%s", version.majorMinorConcatenated()),
+			"poetry",
+		},
+		InstallStage: &Stage{
+			Command: "poetry install --no-dev --no-interaction --no-ansi",
+		},
+		BuildStage: &Stage{
+			Command: "poetry build",
+		},
+		// TODO parse pyproject.toml to get the start command?
+		StartStage: &EntrypointStage{
+			Entrypoint:     "python main.py",
+			Image:          fmt.Sprintf("python:%s-alpine", version.exact()),
+			PrepareCommand: "pip install dist/*.whl",
```

Comment:
Sad that this needs to use `pip` but install uses `poetry`. I wish we can have nix runtime as base image so there will be no discrepancy.